docs/adr/adr-0023-adding-stylelint.md

```diff
@@ -0,0 +1,234 @@
+
+## Problems encountered
+
+While we are satisfied with our current configuration we encountered some problems in the process with some rules.
+
+### The whitespace rule
+
+First, we had some problems with the whitespace rule (no-eol-whitespace-rule). The purpose of this rule is not to have whitespace after a line ending (e.g: semicolon or empty lines). The problem with this is that the processor we use targets only the CSS inside template literals which means the backtick wasn't included.
+This leads the code going from this: 
+```javascript
+static get styles () {
+    return [
+      skeletonStyles,
+      // language=CSS
+      css`
+        :host {
+          display: block;
+        }
+      `,
+    ];
+```
+to this:
+```javascript
+static get styles () {
+    return [
+      skeletonStyles,
+      // language=CSS
+      css`
+        :host {
+          display: block;
+        }
+`,
+    ];
+```
+As you can see the ending backtick of the template literal isn't formatted correctly because of this rule as the rule is triggered as an empty line with whitespaces in it.
+As a consequence for now we have deactivated this rule. 
+
+### Selector name case
+
+Another problem was the case on our selectors, indeed by default the `Standard` configuration that we extend wants the selectors to be written as `kebab-case` whereas in our components we also used `kebab--case` and `snake__case`.
+
+To resolve this problem we decided to override these rules (id and class selectors) to match our needs which lead us to have the rule as follows: 
+```json
+{
+   "selector-class-pattern": [
+      "^([a-z][a-z0-9]*)((-{1,2}|_{1,2})[a-z0-9]+)*$",
+      {
+        "message": "Expected class selector to be either kebab-case, kebab--case or snake__case"
+      }
+    ],
+    "selector-id-pattern": [
+      "^([a-z][a-z0-9]*)((-{1,2}|_{1,2})[a-z0-9]+)*$",
+      {
+        "message": "Expected id selector to be either kebab-case, kebab--case or snake__case"
+      }
+    ]
+}
+```
+
+To make it work we just modified the regular expression that is directly provided on the Stylelint website to match our needs and provided a custom message to know precisely what's happening otherwise we would just have had a generic message.
+
+### Duplicate selectors on atoms
+
+On our `atoms` components like for example `cc-button` or `cc-select` while making our own we still use the default HTML element and tweak it. This makes us have the same base selector twice. One for resetting purposes and another one to style it. 
+The problem is that we have a Stylelint rule that doesn't allow duplicate selectors. 
+To resolve this issue and still keep the rule we have decided to put an ignore comment on these selectors.
+
+### No-descending-specificity
+
+[No-descending-specificity](https://stylelint.io/user-guide/rules/no-descending-specificity/) is a rule that forces the order of the selectors within their specificity.
```

Comment:
I like having the link to the official Stylint description. Can you apply that everywhere in this ADR?

Also, you could choose to have those links using `code block` (if it's feasible with markdown syntax).